### PR TITLE
feat: [SPG-340]: add Chaos step in PipelineStudio

### PR DESCRIPTION
**Pull request by abhinavrastogi-harness**
### Summary

Adds init code for Chaos Step in PipelineStudio

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 69 Issues 
* Code issue 69 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11549)

**Comment by abhinavrastogi-harness** [comment]
Current State:

Screenshot 2022-09-29 at 5 49 56 PM
Screenshot 2022-09-29 at 5 50 02 PM

**Comment by arkajyotiMukherjee** [comment]
Current Progress:
image

image

File: src/modules/70-pipeline/components/PipelineStudio/StepPalette/StepPalette.tsx [hunk 1]
```diff
@@ -100,7 +102,13 @@ export function StepPalette({ onSelect, stepsFactory, stepPaletteModuleInfos }:
     })
 
     if (toShow) {
-      setStepsCategories(toShow)
+      // Hide Chaos step when CHAOS_ENABLED is false
+      if (!CHAOS_ENABLED) {
+        const filterSteps = toShow.filter(step => step.name !== 'Chaos')
```
Comment (abhinavrastogi-harness) [inline src/modules/70-pipeline/components/PipelineStudio/StepPalette/StepPalette.tsx:107]: BE should not be sending the Chaos step in the first place if FF is false.
we dont have conditions for any other module here... checks for all steps are in BE only.

Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/components/PipelineStudio/StepPalette/StepPalette.tsx:107]: Yeah, I have already conveyed this to @arkajyotiMukherjee , and he said he will ask to BE to do this from their side.

Comment (kapilkumawat86) [reply src/modules/70-pipeline/components/PipelineStudio/StepPalette/StepPalette.tsx:107]: this support is there in BE, ask them to check with archit

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/PipelineStepInterface.ts [hunk 2]
```diff
@@ -117,7 +117,8 @@ export enum StepType {
   CustomDeployment = 'CustomDeployment',
   FetchInstanceScript = 'FetchInstanceScript',
   Wait = 'Wait',
-  ShellScriptProvision = 'ShellScriptProvision'
+  ShellScriptProvision = 'ShellScriptProvision',
+  ChaosExperiment = 'Chaos' //todoChaos
```
Comment (mayankVermaHarness) [inline src/modules/70-pipeline/components/PipelineSteps/PipelineStepInterface.ts:121]: Why is there a `todo`? Anything pending here?

Comment (mayankVermaHarness) [inline src/modules/70-pipeline/components/PipelineSteps/PipelineStepInterface.ts:121]: Why is there a `todo`? Anything pending here?

File: src/modules/70-pipeline/components/PipelineStudio/StepPalette/StepPalette.tsx (hunk 1 above)
Comment (mayankVermaHarness) [inline src/modules/70-pipeline/components/PipelineStudio/StepPalette/StepPalette.tsx:107]: Is there a constant for the string `'Chaos'`?

File: src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentExecutionView/ChaosExperimentExecView.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    nodeExecutionId: step.uuid || /* istanbul ignore next */ ''
+  })
+  const { showError } = useToaster()
+  const { getString } = useStrings()
+  const { getRBACErrorMessage } = useRBACError()
+  const history = useHistory()
+
+  function handleChange(strategy: Strategy): void {
+    const interruptType = strategy as HandleManualInterventionInterruptQueryParams['interruptType']
+    handleInterrupt(undefined, {
```
Comment (mayankVermaHarness) [inline src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentExecutionView/ChaosExperimentExecView.tsx:66]: Do we want to pass `null` instead of `undefined`?

File: src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentExecutionView/ChaosExperimentExecView.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    <Layout.Horizontal spacing={'small'} className={cx({ [css.loading]: loading })}>
+      {isManualInterruption ? (
+        <Popover
+          minimal={true}
+          position={Position.BOTTOM_RIGHT}
+          content={
+            <Menu>
+              {STRATEGIES.map((layer, i) => {
+                return (
+                  <div key={i} className={css.listStyle}>
```
Comment (mayankVermaHarness) [inline src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentExecutionView/ChaosExperimentExecView.tsx:96]: Can we use `id` as key instead of index?

File: src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentExecutionView/ChaosExperimentExecView.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+                    {layer.map((strategy, j) => (
+                      <Menu.Item
+                        key={j}
```
Comment (mayankVermaHarness) [inline src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentExecutionView/ChaosExperimentExecView.tsx:99]: Same goes here: Can we use `id` as key instead of index?

File: src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentStep.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,213 @@
+/*
```
Comment (mayankVermaHarness) [inline src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentStep.stories.tsx:1]: Is this file intended? Are `stories` intended to be put in `NG-UI` codebase?

File: src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentStepFunctionConfigs.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  {
+    name: 'spec.expectedResilienceScore',
+    type: ValidationFieldTypes.Numeric,
+    label: 'chaos.pipelineStep.expectedResiliencyScoreLabel',
+    isRequired: true
+  }
+]
+
+export function getInputSetViewValidateFieldsConfig(): Array<{
+  // isRequired = true
```
Comment (mayankVermaHarness) [inline src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentStepFunctionConfigs.ts:62]: Remove comment. Another optional `isRequired` is present on Line 66

File: src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentStepFunctionConfigs.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  name: string
+  type: ValidationFieldTypes
+  label?: string
+  isRequired?: boolean
+}> {
+  return []
```
Comment (mayankVermaHarness) [inline src/modules/75-chaos/components/PipelineSteps/ChaosExperimentStep/ChaosExperimentStepFunctionConfigs.ts:68]: This function returns an empty array. Is this intended?

**Review by mayankVermaHarness — APPROVED** [review]
Added a few comments, but nothing seems to be a blocker. Hence, approving.

**Comment by arkajyotiMukherjee** [comment]
We'll take all the review comments in a subsequent PR. due to some tight deadlines, we had to merge this one. @mayankVermaHarness @kapilkumawat86 thanks for the reviews.
